src/DynamoCoreWpf/ViewModels/Preview/WatchHoverIconViewModel.cs

```diff
@@ -0,0 +1,212 @@
+            PlaceWatchNode();
+        }
+
+        bool CanExecute(object parameter)
+        {
+            return true;
+        }
+
+        void InitCommands()
+        {
+            PlaceWatchNodeCommand = new DelegateCommand(PlaceWatchNodeCommandExecute,CanExecute);
+        }
+
+        public WatchHoverIconViewModel(ConnectorViewModel connectorViewModel, DynamoViewModel dynamoViewModel)
+        {
+            ViewModel = connectorViewModel;
+            DynamoViewModel = dynamoViewModel;
+            InitCommands();
+
+            Dispatcher = Dispatcher.CurrentDispatcher;
+
+            var bez = ViewModel.BezierControlPoints;
+            var coll = ViewModel.ConnectorPinViewCollection;
+
+            if (ViewModel.ConnectorPinViewCollection.Count==0 && ViewModel.BezierControlPoints is null)
+                MidPoint = ConnectorBezierMidpoint();
+            else
+                MidPoint = MultiBezierMidpoint();
+
+            connectorViewModel.PropertyChanged += OnConnectorViewModelPropertyChanged;
+            IsHalftone = false;
+        }
+
+        private void OnConnectorViewModelPropertyChanged(object sender, System.ComponentModel.PropertyChangedEventArgs e)
+        {
+            if (!(e.PropertyName.Contains("CurvePoint") |
+                  e.PropertyName == nameof(ConnectorViewModel.ConnectorPinViewCollection)))
+                return;
+            
+                if (ViewModel.ConnectorPinViewCollection.Count > 0 && ViewModel.BezierControlPoints != null)
+                    MidPoint = MultiBezierMidpoint();
+                else
+                    MidPoint = ConnectorBezierMidpoint();
+
+                RaisePropertyChanged(nameof(MidPoint));
+            
+        }
+
+        private Point ConnectorBezierMidpoint()
+        {
+            // formula to get bezier curve midtpoint
+            // https://stackoverflow.com/questions/5634460/quadratic-b%c3%a9zier-curve-calculate-points?rq=1
+            var parameter = 0.5;
+            var x = ((1 - parameter) * (1 - parameter) * (1 - parameter) *
+                ViewModel.CurvePoint0.X + 3 * (1 - parameter) * (1 - parameter)
+                * parameter *
+                ViewModel.CurvePoint1.X + 3 * (1 - parameter)
+                * parameter * parameter *
+                ViewModel.CurvePoint2.X + parameter * parameter * parameter *
+                ViewModel.CurvePoint3.X) - (MarkerSize / 2);
+
+            var y = ((1 - parameter) * (1 - parameter) * (1 - parameter) *
+                ViewModel.CurvePoint0.Y + 3 * (1 - parameter) * (1 - parameter)
+                * parameter *
+                ViewModel.CurvePoint1.Y + 3 * (1 - parameter)
+                * parameter * parameter *
+                ViewModel.CurvePoint2.Y + parameter * parameter * parameter *
+                ViewModel.CurvePoint3.Y) - (MarkerSize / 2);
+
+            return new Point(x, y);
+        }
+
+        private Point ConnectorBezierMidpoint(Point[] points)
```

Comment:
why do we have two of these?

Reply:
One was the OOTB one, using the `CurvePoint0, CurvePoint1` etc.. The second one takes an array of points as argument which enable use to draw bezier segments between multiple points. Removing the hard-wired one, keeping the arguments one.